src/shapes/Object/StackedObject.ts

```diff
@@ -113,41 +123,41 @@ export class StackedObject<
           ...otherAncestors.slice(0, otherAncestors.length - 1),
         ],
         common: [this],
-      } as AncestryComparison<S>;
+      };
     }
     //  compare ancestors
     for (let i = 0, ancestor; i < ancestors.length; i++) {
       ancestor = ancestors[i];
-      if (ancestor === other) {
+      if (isGroup(other) && ancestor === other) {
         return {
           fork: [this, ...ancestors.slice(0, i)],
           otherFork: [],
           common: ancestors.slice(i),
-        } as AncestryComparison<S>;
+        };
       }
       for (let j = 0; j < otherAncestors.length; j++) {
-        if (this === otherAncestors[j]) {
+        if (isGroup(this) && this === otherAncestors[j]) {
           return {
             fork: [],
             otherFork: [other, ...otherAncestors.slice(0, j)],
-            common: [this, ...ancestors],
-          } as AncestryComparison<S>;
+            common: [this, ...ancestors.slice()],
+          };
         }
         if (ancestor === otherAncestors[j]) {
           return {
             fork: [this, ...ancestors.slice(0, i)],
             otherFork: [other, ...otherAncestors.slice(0, j)],
             common: ancestors.slice(i),
-          } as AncestryComparison<S>;
+          };
         }
       }
     }
     // nothing shared
     return {
-      fork: [this, ...ancestors],
-      otherFork: [other, ...otherAncestors],
+      fork: [this, ...ancestors.slice()],
+      otherFork: [other, ...otherAncestors.slice()],
```

Comment:
what the slices are doing here? why slice + spread?

Reply:
To make TS happy because the `slice` return type will return `Ancestors<S>`